### PR TITLE
[PIP] update manifest to include lib_api.cc

### PR DESCRIPTION
**Pull request by szha**
## Description ##
update manifest to include lib_api.cc

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] update manifest to include lib_api.cc

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: tools/pip/MANIFEST.in [hunk 1]
```diff
@@ -29,4 +29,5 @@ recursive-include mxnet *_LICENSE
 recursive-include mxnet *.h
 recursive-include mxnet *.hpp
 recursive-include mxnet *.cuh
+recursive-include mxnet *.cpp
```
Comment (samskalicky) [inline tools/pip/MANIFEST.in:32]: does *.cpp include *.cc files?
https://github.com/apache/incubator-mxnet/blob/master/src/lib_api.cc

**Comment by samskalicky** [comment]
Which setup.py do we use (or do we want to use) on master moving forward:
https://github.com/apache/incubator-mxnet/blob/master/tools/pip/setup.py
https://github.com/apache/incubator-mxnet/blob/master/python/setup.py

Can we remove the other one?

File: tools/pip/MANIFEST.in (hunk 1 above)
Comment (szha) [reply tools/pip/MANIFEST.in:32]: actually no, let me include cc as well.

**Comment by szha** [comment]
@samskalicky I think both need to be kept for now. The needs for packaging is quite different from setting up the package locally.

**Comment by samskalicky** [comment]
> @samskalicky I think both need to be kept for now. The needs for packaging is quite different from setting up the package locally.

makes sense, so the one in tools/pip/setup.py is the one we use for building wheels, everywhere (dist.mxnet.io, releases, etc)?

File: tools/pip/MANIFEST.in (hunk 1 above)
Comment (samskalicky) [reply tools/pip/MANIFEST.in:33]: cool, were you able to package the wheel and unzip it and confirm the lib_api.cc was inside?

**Comment by samskalicky** [comment]
everything else looks good to me, can you confirm packaging the wheel now includes the lib_api.cc?

File: tools/pip/setup.py [hunk 2]
```diff
@@ -103,6 +103,7 @@
                 os.path.join(CURRENT_DIR, 'mxnet/include/nnvm'))
 
 # copy cc file for mxnet extensions
+os.mkdir(os.path.join(CURRENT_DIR, 'mxnet/src'))
 shutil.copy(os.path.join(CURRENT_DIR, 'mxnet-build/src/lib_api.cc'),
             os.path.join(CURRENT_DIR, 'mxnet/src'))
```
Comment (szha) [inline tools/pip/setup.py:108]: without creating src folder, the semantics of this is to create a regular file called `src` with the content of lib_api.cc...

**Comment by szha** [comment]
And now we are good.

**Review by samskalicky — APPROVED** [review]
Thanks @szha !

**Comment by samskalicky** [comment]
@mxnet-bot run ci [unix-cpu]

**Pull request merged**